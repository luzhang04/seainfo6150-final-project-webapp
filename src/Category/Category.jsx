import React from 'react'
import styles from "./Category.module.css";
import { Link } from "react-router-dom";
import easter from "../images/easter.jpeg";
import halloween from "../images/halloween.jpeg";
import stpatricks from "../images/stpatricks.jpeg";
import valentines from "../images/valentines.jpeg";
import christmas from "../images/christmas.jpeg";
import PropTypes from "prop-types";


const Category = () => {

  return (
    <div className={styles.container}>
    <ul className={styles.ul}>
      <li className={styles.list}>
        <Link className={styles.diylink} to={`/diylist/easter`}>Easter DIY</Link>
        <Link to={`/diylist/easter`} ><img src={easter} alt="easter diy"/></Link>
      </li>
      <li className={styles.list}>
      <Link className={styles.diylink} to={`/diylist/halloween`}>Halloween DIY</Link>
      <Link to={`/diylist/halloween`} ><img src={halloween} alt="halloween diy"/></Link>
      </li>
      <li className={styles.list}>
      <Link className={styles.diylink} to={`/diylist/st-patricks`}>St Patricks DIY</Link>
      <Link to={`/diylist/st-patricks`} ><img src={stpatricks} alt="st-patricks diy"/></Link>
      </li>
      <li className={styles.list}>
      <Link className={styles.diylink} to={`/diylist/valentines`}>Valentines DIY</Link>
      <Link to={`/diylist/valentines`} ><img src={valentines} alt="valentines diy"/></Link>
      </li>
      <li className={styles.list}>
      <Link className={styles.diylink} to={`/diylist/christmas`}>Christmas DIY</Link>
      <Link to={`/diylist/christmas`} ><img src={christmas} alt="christmas diy"/></Link>
      </li>
    </ul>
    <br />

 

    </div>
  );
};

Category.propTypes = {
    crafts: PropTypes.array.isRequired
}

export default Category;
