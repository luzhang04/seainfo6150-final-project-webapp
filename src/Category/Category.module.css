.container {
	background: #edf6f9;
	border: 1px solid #edf6f9;
	font-family: 'Montserrat', Helvetica, sans-serif;
	display: grid;
	grid-template-columns: auto;
	align-items: center;
	gap: 20px;
	list-style-type: none;
	padding: 150px;
	justify-items: center;
}

.ul {
	display: grid;
	padding: 15px;
	gap: 20px;
}

.list {
	display: grid;
	grid-template-columns: 1fr;
	background: white;
	border: 10px solid #edf6f9;
	border-radius: 10px;
	box-shadow: 0px 0px 10px 5px #a8dadc;
	gap: 20px;
	padding: 15px;
}
.diylink {
	color: #457b9d;
	text-decoration: none;
	font-size: 2em;
}

a:hover {
	color: #fab1a0;
}

@media screen and (min-width: 200px) {
	.ul {
		grid-template-columns: repeat(1, 1fr);
	}
	.container {
		padding: 50px;
	}
}

@media screen and (min-width: 800px) {
	.ul {
		grid-template-columns: repeat(2, 1fr);
	}
	.container {
		padding: 100px;
	}
}

@media screen and (min-width: 1100px) {
	.ul {
		grid-template-columns: repeat(3, 1fr);
	}
	.diylink {
		font-size: 1.5em;
	}
	.container {
		padding: 150px;
	}
}

@media screen and (min-width: 1500px) {
	.ul {
		grid-template-columns: repeat(4, 1fr);
	}
	.diylink {
		font-size: 2em;
	}
}
