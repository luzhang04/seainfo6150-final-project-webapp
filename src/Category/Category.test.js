import React from 'react';
import { render } from '@testing-library/react';
import Category from './Category.jsx';
import { BrowserRouter } from "react-router-dom";

describe('Category tests', () => {
	it('renders correctly', () => {
		const { container } = render(<BrowserRouter><Category /></BrowserRouter>);
		expect(container).toMatchSnapshot();
	});
});