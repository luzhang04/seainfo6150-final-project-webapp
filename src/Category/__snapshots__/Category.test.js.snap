// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <ul
      class="ul"
    >
      <li
        class="list"
      >
        <a
          class="diylink"
          href="/diylist/easter"
        >
          Easter DIY
        </a>
        <a
          href="/diylist/easter"
        >
          <img
            alt="easter diy"
            src="easter.jpeg"
          />
        </a>
      </li>
      <li
        class="list"
      >
        <a
          class="diylink"
          href="/diylist/halloween"
        >
          Halloween DIY
        </a>
        <a
          href="/diylist/halloween"
        >
          <img
            alt="halloween diy"
            src="halloween.jpeg"
          />
        </a>
      </li>
      <li
        class="list"
      >
        <a
          class="diylink"
          href="/diylist/st-patricks"
        >
          St Patricks DIY
        </a>
        <a
          href="/diylist/st-patricks"
        >
          <img
            alt="st-patricks diy"
            src="stpatricks.jpeg"
          />
        </a>
      </li>
      <li
        class="list"
      >
        <a
          class="diylink"
          href="/diylist/valentines"
        >
          Valentines DIY
        </a>
        <a
          href="/diylist/valentines"
        >
          <img
            alt="valentines diy"
            src="valentines.jpeg"
          />
        </a>
      </li>
      <li
        class="list"
      >
        <a
          class="diylink"
          href="/diylist/christmas"
        >
          Christmas DIY
        </a>
        <a
          href="/diylist/christmas"
        >
          <img
            alt="christmas diy"
            src="christmas.jpeg"
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
